sycl/source/handler.cpp

```diff
@@ -440,6 +476,46 @@ void handler::extractArgsAndReqsFromLambda(
     const detail::kernel_param_desc_t *KernelArgs, bool IsESIMD) {
   const bool IsKernelCreatedFromSource = false;
   size_t IndexShift = 0;
+
+  // TODO: to do that in compile time.
```

Comment:
At the very least, please reuse the code above

Reply:
Unfortunately, the code above is different from that, so we can't reuse them.
Here are some of them: `if (!IsKernelCreatedFromSource) {` vs `if (!IsESIMD) {`  
 `UnPreparedArgs[I].MSize` vs `KernelArgs[I].info` 
 `UnPreparedArgs[I].MType` vs `KernelArgs[I].kind`
and the difference in `case access::target::local`

Reply:
Differences can be handled by a function arguments.